pimd/pim_addr.h

```diff
@@ -27,13 +27,19 @@
 
 #if PIM_IPV == 4
 typedef struct in_addr pim_addr;
-
+typedef struct sockaddr_in pimsock_in;
+typedef struct in_pktinfo pim_af_pkt_info;
 #define PIM_ADDRSTRLEN	INET_ADDRSTRLEN
 #define PIM_AF		AF_INET
 #define PIM_AFI		AFI_IP
 #define PIM_MAX_BITLEN	IPV4_MAX_BITLEN
 #define PIM_AF_NAME     "ip"
 #define	FRR_PIM_AF_XPATH_VAL "frr-routing:ipv4"
+#define PIM_AF_OPT      IPPROTO_IP
+#define PIM_AF_MCAST_PKTINFO  IP_PKTINFO
+#define PIM_AF_MCAST_TTL      IP_MULTICAST_TTL
+#define PIM_AF_MCAST_IF       IP_MULTICAST_IF
+#define PIM_AF_MCAST_LOOP     IP_MULTICAST_LOOP
```

Comment:
Since these definitions are only used in `pim_sock.c` I think we should move them there.

(That also makes the code in `pim_sock.c` a bit more readable since the definitions are in the same place)

Reply:
Ack

Reply:
I have moved the definitions to pim_sock.c for this PR. But the pim_ssmpingd.c changes which we are in the process of doing would require these macros too. Can we move these to pim_sock.h ?